### ISSUE TITLE
Egdetpu.tflite Converting Error

### ISSUE THREAD

**Issue by Zee0549** [google-coral/edgetpu#416]
Hi there,

When I try to convert my .tflite model to Edgetpu.tflite, I got this error:

’‘’
Edge TPU Compiler version 15.0.340273435
Invalid model: model_nasa_bearing_CNN.tflite
Model not quantized
‘’‘

Here is how I convert my tf model to tflite:

‘’‘
run_model = tf.function(lambda x: model(x))
BATCH_SIZE = 1
STEPS = 4
INPUT_SIZE = 1279
concrete_func = run_model.get_concrete_function(tf.TensorSpec([BATCH_SIZE, STEPS, INPUT_SIZE], model.inputs[0].dtype))

MODEL_DIR = "/content/gdrive/MyDrive/Colab Notebooks/tf_saved_model"
model.save(MODEL_DIR, save_format="tf", signatures=concrete_func)

converter = tf.lite.TFLiteConverter.from_saved_model(MODEL_DIR)
tflite_model = converter.convert()
‘’‘

I also tried: 
’‘’
tf.lite.TFLiteConverter.from_keras_model()
and 
tf.lite.TFLiteConverter.from_saved_model(MODEL_DIR) 
without using concrete function
‘’‘
The error is still the same.

Do you have any idea about what happened there?

https://drive.google.com/file/d/1ivl5h0X803EsGjRLFXswRBQbjCxwRye4/view?usp=sharing   Here is my Demo LSTM tflite model.

Thank you in advance.
Ze

**Comment by hjonnala** [comment]
Hello, 

I 've visualized the model using netron.app. I can see the model layers have datatype of float 32. 
we need to fully quantize the model to int8 format.

please check these tutorial section convert to TF Lite
https://colab.sandbox.google.com/github/google-coral/tutorials/blob/master/retrain_classification_ptq_tf2.ipynb#scrollTo=P0_StAtZwJ5p

**Comment by Zee0549** [comment]
Hi,

I followed this instruction, but I got this error when I am converting LSTM:

ValueError: Failed to parse the model: pybind11::init(): factory function returned nullptr.

The weird thing is, when I created a CNN model, it works well. Does this mean LSTM is not supported yet?

This is how I create this demo LSTM model:

model = tf.keras.models.Sequential([
    tf.keras.layers.Input(shape=(4, 1279), name='input'),
    tf.keras.layers.LSTM(20, time_major=False, return_sequences=True),
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(2, activation='softmax', name='output')
])
model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])
model.summary()

**Comment by hjonnala** [comment]
can you please share your quantization script and saved model?

**Comment by Zee0549** [comment]
Hi,
Here is the Colab link:
https://colab.research.google.com/drive/1_NstPsGiyZE5VqnRREDPwUvrV6Ethqma?usp=sharing

**Comment by hjonnala** [comment]
There is an open issue in tensorflow repository for this error.

https://github.com/tensorflow/tensorflow/issues/46821

please check with tensorflow team.

**Comment by Zee0549** [comment]
Hi, 

Thanks a lot for your support. I will check this with tf team.

BTW, I got this edgetpu.tflite works on the google dev board now. I noticed that, to run an inference with edgetpu.tflite, the data type should be uint8. However, while creating the tf model, I normalized my data to range (0, 1). In this case, if I convert my data to uint8, all my data will be 0 or 1. Does this mean I can not normalize my data to (0, 1) anymore if I want to use edgetpu.tflite?
